discovery-provider/src/queries/get_users.py

```diff
@@ -1,59 +1,93 @@
 import logging # pylint: disable=C0302
 from sqlalchemy import asc
 
+from flask.globals import request
 from src import exceptions
 from src.models import User
 from src.utils import helpers
 from src.utils.db_session import get_db_read_replica
-from src.queries.query_helpers import get_current_user_id, populate_user_metadata, paginate_query
+from src.queries.query_helpers import populate_user_metadata, paginate_query
+from src.utils.redis_cache import extract_key, use_redis_cache
 
 logger = logging.getLogger(__name__)
 
+UNPOPULATED_USER_CACHE_DURATION_SEC = 10
+
+def make_cache_key(args):
+    cache_keys = {}
+
+    if args.get("id"):
+        ids = args.get("id")
+        ids = map(str, ids)
+        ids = ",".join(ids)
+        cache_keys["user_id"] = ids
+
+    if args.get("handle"):
+        handle = args.get("handle")
+        cache_keys["handle"] = handle
+
+    if args.get("wallet"):
+        wallet = args.get("wallet")
+        cache_keys["wallet"] = wallet
+
+    key = extract_key(f"unpopulated-user:{request.path}", cache_keys.items())
```

Comment:
gonna make a pass later to remove the request path from these?

Reply:
yeah i think we should